### PR TITLE
hotfix in profile view model

### PR DESCRIPTION
**Pull request by lucamouchel**
Fixes a problem when we searched for users it displayed the current user

### CODE REVIEW

**Review by TheGuyWithoutH — APPROVED** [review]
Good to go!

**Review by alberttkt — CHANGES_REQUESTED** [review]
Nice, just a comment

File: app/src/main/java/ch/epfl/culturequest/ui/profile/ProfileViewModel.java [hunk 1]
```diff
@@ -29,32 +30,20 @@ public ProfileViewModel(String uid) {
...
-                username.postValue(p.getUsername());
-                profilePictureUri.postValue(p.getProfilePicture());
-                pictures.postValue(p.getImagesList());
+            p.addObserver((profileObject, arg) -> {
+                Profile profile = (Profile) profileObject;
+                username.postValue(profile.getUsername());
+                profilePictureUri.postValue(profile.getProfilePicture());
+                pictures.postValue(profile.getImagesList());
             });
+        });
```
[the remark this answers is not in the document]
Comment (lucamouchel) [reply app/src/main/java/ch/epfl/culturequest/ui/profile/ProfileViewModel.java]: yes you're right, i rolled back to the previous code and just added a check

**Review by alberttkt — APPROVED** [review]
Nice!

File: app/src/main/java/ch/epfl/culturequest/ui/profile/ProfileViewModel.java [hunk 2]
```diff
@@ -30,33 +31,33 @@ public ProfileViewModel(String uid) {
...
-                pictures.postValue(p.getImagesList());
-            });
-
+            if (selectedProfile != null
+                    && selectedProfile.getUid().equals(uid)) {
+                Database.getProfile(uid).whenComplete((p, e) -> {
+                    username.setValue(p.getUsername());
+                    profilePictureUri.setValue(p.getProfilePicture());
+                    pictures.setValue(p.getImagesList());
+                });
```
Comment (alberttkt) [inline app/src/main/java/ch/epfl/culturequest/ui/profile/ProfileViewModel.java:43]: why do you need to fetch the profile from the database if you already have selectedProfile?

**Review by alberttkt — CHANGES_REQUESTED** [review]

**Pull request merged**